### PR TITLE
Update netci and run-test to use the IgnoreForCI trait in Jenkins

### PR DESCRIPTION
**Pull request by ianhays**
Updates the netci.groovy and runtest.sh files to pass -notrait Category=IgnoreForCI to build.cmd and run-test.sh in Jenkins so that we can disable tests from running when they're being run in Jenkins.

@dotnet-bot test ci please

replaces #6522 which replaced part of #6145 (I will merge this someday, you know )

@Priya91

### CODE REVIEW

**Comment by ianhays** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -323,7 +323,7 @@ def outerloopLinuxOSes = ['Ubuntu15.10', 'CentOS7.1', 'OpenSUSE13.2', 'RHEL7.2']
             def newJob = job(Utilities.getFullJobName(project, newJobName, isPR)) {
                 steps {
                     if (os == 'Windows 10' || os == 'Windows 7' || os == 'Windows_NT') {
-                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && Build.cmd /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true")
+                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && Build.cmd /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true p:WithoutCategories=IgnoreForCI")
                     }
                     else if (os == 'OSX') {
                         shell("HOME=\$WORKSPACE/tempHome ./build.sh /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true /p:TestWithLocalLibraries=true")
```
Comment (Priya91) [inline netci.groovy]: We should add for build.sh as well no? Innerloop Ubuntu and OSX use build.sh now..

**Comment by Priya91** [comment]
LGTM !

File: netci.groovy (hunk 1 above)
Comment (ianhays) [inline netci.groovy]: Oh, I wasn't aware of that. We should add it for build.sh, yes.

File: netci.groovy [hunk 2]
```diff
@@ -323,7 +323,7 @@ def outerloopLinuxOSes = ['Ubuntu15.10', 'CentOS7.1', 'OpenSUSE13.2', 'RHEL7.2']
             def newJob = job(Utilities.getFullJobName(project, newJobName, isPR)) {
                 steps {
                     if (os == 'Windows 10' || os == 'Windows 7' || os == 'Windows_NT') {
-                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && Build.cmd /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true")
+                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && Build.cmd /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true p:WithoutCategories=IgnoreForCI")
                     }
```
Comment (Priya91) [inline netci.groovy]: this should be ```/p```, also need to be added for outerloop build.sh runs

Comment (ianhays) [inline netci.groovy]: Fixed. Also added it for the build.sh calls.

**Comment by ianhays** [comment]
@dotnet-bot test ci please

**Comment by ianhays** [comment]
@dotnet-bot test ci please

**Comment by ianhays** [comment]
cc: @mmitche

**Pull request merged**